src/components/NameView.tsx

```diff
@@ -85,6 +86,28 @@ export default function NameView (props: Props) {
     }
   }
 
+  const handlePastSpeaker = (e, data) => {
+    dispatch(UnlockBadgeAction(data.badge))
```

Comment:
Have you tested this in a local web client? IIRC `UnlockBadgeAction` is responsible for showing the local user the "hey, you unlocked a new badge!" modal and showing that badge in the selection UI — I don't think this will trigger a network request to server-side unlock a badge for a different user.

Reply:
Ah dang, I think you're right. Which would explain why there's no user ID involved